src/lem/circuit.rs

```diff
@@ -552,26 +646,194 @@ impl Func {
                         let c = AllocatedPtr::from_parts(tag, c_num);
                         bound_allocations.insert(tgt.clone(), c);
                     }
-                    Op::Div(_tgt, _a, _b) => {
-                        // TODO
+                    Op::Div(tgt, a, b) => {
+                        let a = bound_allocations.get(a)?;
+                        let b = bound_allocations.get(b)?;
+                        let a_num = a.hash();
+                        let b_num = b.hash();
+
+                        let b_is_zero = &alloc_is_zero(&mut cs.namespace(|| "b_is_zero"), b_num)?;
+                        let one = g.global_allocator.get_or_alloc_const(cs, F::ONE)?;
+
+                        let divisor = pick(
+                            &mut cs.namespace(|| "maybe-dummy divisor"),
+                            b_is_zero,
+                            &one,
+                            b_num,
+                        )?;
+
+                        let quotient = div(&mut cs.namespace(|| "quotient"), a_num, &divisor)?;
+
+                        let tag = g
+                            .global_allocator
+                            .get_or_alloc_const(cs, Tag::Expr(Num).to_field())?;
+                        let c = AllocatedPtr::from_parts(tag, quotient);
+                        bound_allocations.insert(tgt.clone(), c);
+                    }
+                    Op::Lt(tgt, a, b) => {
+                        let a = bound_allocations.get(a)?;
+                        let b = bound_allocations.get(b)?;
+                        let tag = g
+                            .global_allocator
+                            .get_or_alloc_const(cs, Tag::Expr(Num).to_field())?;
+                        let (preallocated_preimg, lt) =
+                            &g.preallocated_less_than_slots[next_slot.consume_less_than()];
+                        for (i, n) in [a.hash(), b.hash()].into_iter().enumerate() {
+                            implies_equal(
+                                &mut cs.namespace(|| {
+                                    format!("implies equal for component {i} (OP {:?})", &op)
+                                }),
+                                not_dummy,
+                                n,
+                                &preallocated_preimg[i],
+                            )?;
+                        }
+                        let c = AllocatedPtr::from_parts(tag, lt.clone());
+                        bound_allocations.insert(tgt.clone(), c);
+                    }
+                    Op::BitAnd(tgt, a, b) => {
+                        let a = bound_allocations.get(a)?;
+                        let a_bits = a.hash().to_bits_le(&mut cs.namespace(|| "bitwise_and"))?;
```

Comment:
Please justify for me why this does not need to be a strict bit decomposition.

Reply:
Indeed `to_bits_le_strict` is needed here. It is fixed.